siliconcompiler/tools/vpr/place.py

```diff
@@ -24,6 +26,39 @@ def setup(chip, clobber=True):
     chip.add('tool', tool, 'task', task, 'output', design + '.place', step=step, index=index)
 
 
+################################
+# Pre_process (pre executable)
+################################
+
+
+def pre_process(chip):
+    ''' Tool specific function to run before step execution
+    '''
+
+    design = chip.top()
+    step = chip.get('arg', 'step')
+    index = chip.get('arg', 'index')
+
+    if 'pins' in chip.getkeys('input', 'constraint'):
+        chip.logger.info("Using pin constraints file instead of pin constraints from chip.set")
+    else:
+        all_component_constraints = chip.getkeys('constraint', 'component')
+        all_place_constraints = {}
+        if (len(all_component_constraints) > 0):
+            for component in all_component_constraints:
+                place_constraint = chip.get('constraint', 'component', component, 'placement',
+                                            step=step, index=index)
+                chip.logger.info(f'Place constraint for {component} at {place_constraint}')
+                all_place_constraints[component] = place_constraint
+
+            constraints_xml = generate_vpr_constraints_xml(all_place_constraints)
+            xml_file = f'inputs/{design}_constraints.xml'
```

Comment:
use yosys convention for file naming: sc_constraints.xml (the design name is not needed).
also, use the set the path in the schema like: https://github.com/siliconcompiler/siliconcompiler/blob/7516ab2a6108b788aac41a0aa50423132058e4ea/siliconcompiler/tools/yosys/syn_asic.py#L500